engine/options.go

```diff
@@ -372,6 +372,22 @@ func WithRetrievalAddrs(addrs ...string) Option {
 	}
 }
 
+// WithRetrievalPeerID sets the peerID of where to get the content corresponding to an indexing advertisement.
+// If unspecified, the libp2p host peer ID are used.
+// See: WithHost
+func WithRetrievalPeerID(peerID string) Option {
+	return func(o *options) error {
+		if len(peerID) != 0 {
+			pId, err := peer.Decode(peerID)
+			if err != nil {
+				return fmt.Errorf("bad peer ID %q: %w", peerID, err)
+			}
+			o.provider.ID = pId
+		}
+		return nil
+	}
```

Comment:
This functionality is already available by using the `WithProvider` and `WithPrivateKey` options. The `WithProvider` option sets the peer and addresses for the provider to put in indexing advertisements. This is different than `WithHost` which sets the host that is publishing the advertisements (libp2p server).

Setting the provider ID should also be accompanied by a call to `WithPrivateKey`, to set the private key associated with the provider peer ID being set. This way the advertisements will be signed with the provider's key.

The only reason this PR currently works is because the indexer is currently configured to allow any ID to publish on behalf of a provider. This means the advertisement will be accepted if it has a valid signature from anyone. This configuration is long overdue for being locked down so that the indexer only accepts advertisements signed by the provider or from a specific list of delegated publishers.